src/Symfony/Component/Routing/Tests/RouteTest.php

```diff
@@ -67,19 +66,16 @@ public function testOptions()
         $this->assertEquals(array_merge([
         'compiler_class' => 'Symfony\\Component\\Routing\\RouteCompiler',
         ], ['foo' => 'bar']), $route->getOptions(), '->setOptions() sets the options');
-        $this->assertEquals($route, $route->setOptions([]), '->setOptions() implements a fluent interface');
 
         $route->setOptions(['foo' => 'foo']);
         $route->addOptions(['bar' => 'bar']);
-        $this->assertEquals($route, $route->addOptions([]), '->addOptions() implements a fluent interface');
         $this->assertEquals(['foo' => 'foo', 'bar' => 'bar', 'compiler_class' => 'Symfony\\Component\\Routing\\RouteCompiler'], $route->getOptions(), '->addDefaults() keep previous defaults');
     }
 
     public function testOption()
     {
-        $route = new Route('/{foo}');
-        $this->assertFalse($route->hasOption('foo'), '->hasOption() return false if option is not set');
-        $this->assertEquals($route, $route->setOption('foo', 'bar'), '->setOption() implements a fluent interface');
+        $route = new Route('/{foo}', [], [], ['foo' => 'bar']);
+        $this->assertFalse($route->hasOption('baz'), '->hasOption() return false if option is not set');
```

Comment:
You are changing the meaning of the test here. The test made sure that an option that did not exist before the call to `setOption()` is properly set afterwards. You have removed the call to `setOption()` so we're not testing anymore that `setOption()` actually has an effect other than returning the `$route` object itself.